File: Data/Data/Text.py
import tkinter as tk
import random
import os
from PIL import Image, ImageTk
from tkinter import Tk, Label, Button


def close_window():
    window.destroy()
    os.system("python Data/Data/main.py")

translations = [
    "Hello, how are you?",  # Пример предложений на английском
    "What is your name?",
    "Where are you from?",
    "I love programming!",
    "Have a nice day!"
]

sentences = [
    "Привет, как дела?",  # Примеры соответствующих переводов на русский
    "Как тебя зовут?",
    "Откуда ты?",
    "Я люблю программирование!",
    "Хорошего дня!"
]

currentSentenceIndex = 0
currentSentence = ""
currentTranslation = ""

window = tk.Tk()
window.state('zoomed')
window.title("Accelingvo")
window.iconbitmap('Data/py.ico')
window.geometry("1920x1080")
image = Image.open("Data/background.jpg")  # Замените на путь к вашему фоновому изображению
image = image.resize((window.winfo_screenwidth(), window.winfo_screenheight()))
background_image = ImageTk.PhotoImage(image)
labell = Label()
labell.pack()
labell.configure(image=background_image)
labell.place(relx=0, rely=0)


def showSentence():
    os.system("clear")
    sentence_label.config(text=currentSentence, font=("Roboto", 32))


def checkTranslation():
    translation = translation_entry.get()
    if translation == currentTranslation:
        result_label.config(text="Correct translate!", fg="green", font=("Roboto", 32))
    else:
        result_label.config(text="Incorrect translate!", fg="red", font=("Roboto", 32))
    translation_entry.delete(0, tk.END)
    getNextSentence()


def getNextSentence():
    global currentSentenceIndex, currentSentence, currentTranslation
    currentSentenceIndex += 1
    if currentSentenceIndex >= len(sentences):
        currentSentenceIndex = 0
    currentSentence = sentences[currentSentenceIndex]
    currentTranslation = translations[currentSentenceIndex]
    showSentence()


random.seed()
currentSentenceIndex = random.randint(0, len(sentences) - 1)
currentSentence = sentences[currentSentenceIndex]
currentTranslation = translations[currentSentenceIndex]
sentence_label = tk.Label(window, text=currentSentence, font=("Roboto", 32))
sentence_label.pack(pady=100)
translation_entry = tk.Entry(window, fg="#000", font=("Roboto", 32))
translation_entry.pack()
check_button = tk.Button(window, text="Check", command=checkTranslation, font=("Roboto", 32))
check_button.pack(pady=50)
result_label = tk.Label(window, text="", font=("Roboto", 32))
result_label.pack()
next_button = tk.Button(window, text="Next sentence", command=getNextSentence, font=("Roboto", 32))
next_button.pack(pady=50)
back_button = Button(window, text="←", font=("Roboto", 32), command=close_window)
back_button.pack(pady=10)
window.mainloop()

File: text.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <ctime>
#include <Windows.h>
#include <conio.h>

using namespace std;

string sentences[] = {
    "Hello, how are you?",            // Пример предложений на английском
    "What is your name?",
    "Where are you from?",
    "I love programming!",
    "Have a nice day!"
};

string translations[] = {
    "Привет, как дела?",             // Примеры соответствующих переводов на русский
    "Как тебя зовут?",
    "Откуда ты?",
    "Я люблю программирование!",
    "Хорошего дня!"
};

int currentSentenceIndex = 0;
string currentSentence;
string currentTranslation;

void showSentence() {
    system("CLS");
    cout << "Translate the following sentence from English to Russian:\n\n";
    cout << currentSentence << endl;
}

void checkTranslation(string input) {
    if (input == currentTranslation) {
        cout << "Correct translation!" << endl;
    } else {
        cout << "Incorrect translation." << endl;
    }
    _getch();
    showSentence();
}

void getNextSentence() {
    currentSentenceIndex++;
    if (currentSentenceIndex >= sizeof(sentences) / sizeof(sentences[0])) {
        currentSentenceIndex = 0;
    }
    currentSentence = sentences[currentSentenceIndex];
    currentTranslation = translations[currentSentenceIndex];
    showSentence();
}

int main() {
    srand(static_cast<unsigned int>(time(0)));

    // Инициализация окна
    HWND console = GetConsoleWindow();
    RECT r;
    GetWindowRect(console, &r);
    MoveWindow(console, r.left, r.top, 800, 400, TRUE);

    currentSentenceIndex = rand() % (sizeof(sentences) / sizeof(sentences[0]));
    currentSentence = sentences[currentSentenceIndex];
    currentTranslation = translations[currentSentenceIndex];

    showSentence();

    while (true) {
        string input;
        getline(cin, input);

        if (input == "check") {
            cout << "Checking translation..." << endl;
            string translation;
            getline(cin, translation);
            checkTranslation(translation);
        } else if (input == "next") {
            cout << "Next sentence..." << endl;
            getNextSentence();
        }
    }

    return 0;
}